CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+cmake_minimum_required(VERSION 3.23)
+project(
+    cppfront
+    LANGUAGES CXX
+    VERSION 0.1.0
+)
+
+option(
+    CPPFRONT_INSTALL_RULES "Enable install rules for cppfront" "${PROJECT_IS_TOP_LEVEL}"
+)
+
+add_executable(cppfront source/cppfront.cpp)
+add_executable(cppfront::cppfront ALIAS cppfront)
+target_compile_features(cppfront PRIVATE cxx_std_20)
+target_sources(
+    cppfront
+    PRIVATE
+    FILE_SET HEADERS
+    BASE_DIRS source
+    FILES source/common.h source/lex.h source/load.h source/parse.h source/sema.h
+)
+
+add_library(cpp2util INTERFACE)
+add_library(cppfront::cpp2util ALIAS cpp2util)
+target_compile_features(cpp2util INTERFACE cxx_std_20)
+target_include_directories(
+    cpp2util INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+)
+target_sources(
+    cpp2util
+    INTERFACE
+    FILE_SET HEADERS
+    BASE_DIRS include
+    FILES include/cpp2util.h
+)
+
+if (NOT PROJECT_IS_TOP_LEVEL)
+    get_directory_property(_CPPFRONT_MAGIC_DIR PARENT_DIRECTORY)
+    include(cmake/CppfrontHelpers.cmake)
+endif ()
+
+if (CPPFRONT_INSTALL_RULES)
+    include(GNUInstallDirs)
+    include(CMakePackageConfigHelpers)
+
+    set(CPPFRONT_INSTALL_CMAKEDIR "${CMAKE_INSTALL_DATADIR}/cmake/cppfront"
+        CACHE STRING "Install location for cppfront's platform-independent CMake files")
+
+    set(CPPFRONT_INSTALL_EXE_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake/cppfront-exe"
+        CACHE STRING "Install location for cppfront's platform-specific CMake files")
+
+    ##
+    # Install rules for cppfront executable (which is naturally platform
+    # dependent)
+
+    install(
+        TARGETS cppfront
+        EXPORT cppfront-exe-targets
+        COMPONENT cppfront
+    )
+
+    install(
+        EXPORT cppfront-exe-targets
+        DESTINATION "${CPPFRONT_INSTALL_EXE_CMAKEDIR}"
+        NAMESPACE cppfront::
+        COMPONENT cppfront
+    )
+
+    configure_package_config_file(
+        cmake/cppfront-exe-config.cmake.in
+        cmake/cppfront-exe-config.cmake
+        INSTALL_DESTINATION "${CPPFRONT_INSTALL_EXE_CMAKEDIR}"
+        PATH_VARS CMAKE_INSTALL_BINDIR
+    )
```

Comment:
I'd avoid using `configure_package_config_file` and just have a simple config file instead. The config file doesn't even use the `PACKAGE_CMAKE_INSTALL_BINDIR` that is set by `PATH_VARS`. The `INSTALL_DESTINATION` variable also embeds an install prefix that is decided at configure time, but since CMake 3.15, that is at odds with `cmake --install ... --prefix ...`. The config scripts needn't be configureable imo.

Reply:
> The config file doesn't even use the `PACKAGE_CMAKE_INSTALL_BINDIR` that is set by `PATH_VARS`.

Yes, it does:

https://github.com/hsutter/cppfront/blob/84982607c2901d9aecb26ef03534136f33337865/cmake/cppfront-exe-config.cmake.in#L1-L7

> The `INSTALL_DESTINATION` variable also embeds an install prefix that is decided at configure time

No, it doesn't. It's used to compute a relative path to _whichever_ prefix is selected down the line. Here is the generated code, so you can see what it actually does:

```
$ cmake -S . -B build -DCMAKE_INSTALL_PREFIX=$PWD/foobar
$ cmake --build build --target install
$ cat foobar/lib/cmake/cppfront-exe/cppfront-exe-config.cmake 
# ...
get_filename_component(PACKAGE_PREFIX_DIR "${CMAKE_CURRENT_LIST_DIR}/../../../" ABSOLUTE)
# ...
include("${CMAKE_CURRENT_LIST_DIR}/cppfront-exe-targets.cmake")

set_and_check(CPPFRONT_EXECUTABLE "${PACKAGE_PREFIX_DIR}/bin/cppfront")
check_required_components(cppfront-exe)
```
As you can see, this does not mention `foobar` at all, and the package is perfectly relocatable.

Reply:
Okay then, I'm just blind and didn't see the variable used.

After a closer inspection of the module's docs, I can certainly say that I totally mixed `INSTALL_PREFIX` and `INSTALL_DESTINATION` up. I also found that `INSTALL_DESTINATION` is not explained in greater detail. [Two sentences](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#:~:text=path%3E%20given%20to-,INSTALL_DESTINATION,-must%20be%20the) and they don't say much :(

Reply:
Also, `CMAKE_EXECUTABLE_SUFFIX` appears to have a `<LANG>` variant, now that I'm looking at your example.